libraries/shared/watcher/event_watcher.go

```diff
@@ -118,6 +118,7 @@ func (watcher *EventWatcher) withRetry(call func() error, expectedErr error, ope
 				logrus.Errorf("error %s logs: %s", operation, err.Error())
 				if consecutiveUnexpectedErrCount > watcher.MaxConsecutiveUnexpectedErrs {
 					errs <- err
+					return err
```

Comment:
I think we might be able to get away with a bare return here - seems like returning the error is probably unnecessary as long as it's being sent up the channel